libpoolprotocols/stratum/EthStratumClient.cpp

```diff
@@ -173,6 +173,16 @@ void EthStratumClient::disconnect()
 	catch (std::exception const& _e) {
 		cwarn << "Error while disconnecting:" << _e.what();
 	}
+
+        if (m_connection.SecLevel() != SecureLevel::NONE) {
```

Comment:
Maybe null both socket pointer without checking SecLevel?

Reply:
sure, makes sense, also whats with ``m_nonsecuresocket``  ?

Reply:
Yes, both of them.